design-principles.md

```diff
@@ -273,13 +277,29 @@ API design
   - String manipulation to generate data or code that will subsequently
     be interpreted or parsed is a bad practice as it leads to code
     injection vulnerabilities.
+    
+  - APIs should not hide expensive operations. 
```

Comment:
This could be misconstrued.  Perhaps instead say:

> If an operation is expensive, that expense should be represented in the API (e.g. by returning a `Future` or a `Stream`).